Diagnostic points to code with no explanation

### Code

```Rust
#![allow(unused)]
fn main() {
    let a = 0;
    let mut b = None;
    move || {
        b = Some(&a);
    };
}
```

### Current output

```Shell
error[E0521]: borrowed data escapes outside of closure
 --> src/main.rs:6:9
  |
4 |     let mut b = None;
  |         ----- `b` declared here, outside of the closure body
5 |     move || {
6 |         b = Some(&a);
  |         ^^^^^^^^^^^^

For more information about this error, try `rustc --explain E0521`.
```

### Desired output

```Shell

```

### Rationale and extra context

When the compiler points to a piece of code, it usually explains how that code is relevant. Here, however, the compiler just silently points there for some reason.

### Other cases

```Rust

```

### Rust Version

```Shell
Reproducible on the playground with version 1.93.0-nightly (2025-11-01 bd3ac0330018c23b111b)
```

### Anything else?

_No response_

--- a/compiler/rustc_borrowck/src/diagnostics/region_errors.rs
+++ b/compiler/rustc_borrowck/src/diagnostics/region_errors.rs
@@ -690,6 +690,17 @@ impl<'infcx, 'tcx> MirBorrowckCtxt<'_, 'infcx, 'tcx> {
             );
 
             diag.span_label(*span, format!("`{fr_name}` escapes the {escapes_from} body here"));
+        } else {
+            diag.span_label(
+                *span,
+                format!("a temporary borrow escapes the {escapes_from} body here"),
+            );
+            if let Some((Some(outlived_name), _)) = outlived_fr_name_and_span {
+                diag.help(format!(
+                    "`{outlived_name}` is declared outside the {escapes_from}, \
+                     so any data borrowed inside the {escapes_from} cannot be stored into it"
+                ));
+            }
         }
 
         // Only show an extra note if we can find an 'error region' for both of the region

--- a/tests/ui/borrowck/borrowed-data-escapes-closure-148392.rs
+++ b/tests/ui/borrowck/borrowed-data-escapes-closure-148392.rs
@@ -1,0 +1,13 @@
+// Test for issue #148392
+// Provides helpful explanations even for anonymous references
+// under the scenario of escaping closure
+
+#![allow(unused)]
+
+fn main() {
+    let a = 0;
+    let mut b = None;
+    move || {
+        b = Some(&a); //~ ERROR borrowed data escapes outside of closure
+    };
+}

--- a/tests/ui/borrowck/borrowed-data-escapes-closure-148392.stderr
+++ b/tests/ui/borrowck/borrowed-data-escapes-closure-148392.stderr
@@ -1,0 +1,14 @@
+error[E0521]: borrowed data escapes outside of closure
+  --> $DIR/borrowed-data-escapes-closure-148392.rs:11:9
+   |
+LL |     let mut b = None;
+   |         ----- `b` declared here, outside of the closure body
+LL |     move || {
+LL |         b = Some(&a);
+   |         ^^^^^^^^^^^^ a temporary borrow escapes the closure body here
+   |
+   = help: `b` is declared outside the closure, so any data borrowed inside the closure cannot be stored into it
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0521`.

